test/test_bools.py

```diff
@@ -18,26 +21,85 @@ class Flags(TestSetup):
+    "flag,expected_f",
+    [
+        ("--a 5", False),
+        ("--a 5 --f", True),
+        ("--a 5 --nof", False),
+        ("--a 5 --f=true", True),
+        ("--a 5 --f true", True),
+        ("--a 5 --f True", True),
+        ("--a 5 --f=false", False),
+        ("--a 5 --f false", False),
+        ("--a 5 --f False", False),
+        ("--a 5 --f --f false", False),
+    ],
+)
+def test_bool_base_work(flag: str, expected_f: bool):
+    assert Base.setup(flag).f is expected_f
 
-    true_strings = ["True", "true"]
-    for s in true_strings:
-        ext = Base.setup(f"--a 5 --f {s}")
-        assert ext.f is True
+@pytest.mark.parametrize(
+    "flag,a,b,c",
+    [
+        ("--a", True, True, False),
+        ("--a true --b --c", True, True, True),
+        ("--a true --noc --b --c --noc", True, True, False),
+        ("--noa --b false --noc", False, False, False),
+    ],
+)
+def test_bool_flags_work(flag: str, a: bool, b: bool, c: bool):
+    flags = Flags.setup(flag)
+    assert flags.a is a
+    assert flags.b is b
+    assert flags.c is c
 
-    false_strings = ["False", "false"]
-    for s in false_strings:
-        ext = Base.setup(f"--a 5 --f {s}")
-        assert ext.f is False
 
+@pytest.mark.parametrize(
+    "flag, nargs, a",
+    [
+        # By default, support both --noflag and --flag=false
+        ("--a", '?', True),
+        ("--noa", '?', False),
+        ("--a true", '?', True),
+        ("--a true false", '?', SystemExit('unrecognized arguments')),
+        # `nargs=None` is like `nargs='?'`
+        ("--a", None, True),
+        ("--noa", None, False),
+        ("--a true", None, True),
+        ("--a true false", None, SystemExit('unrecognized arguments')),
+        # 1 argument explicitly required
+        ("--a", 1, SystemExit('expected 1 argument')),
+        ("--noa", 1, SystemExit('the following arguments are required')),
+        ("--a=true", 1, [True]),
+        ("--a true false", 1, SystemExit('unrecognized arguments')),
+        # 2 argument explicitly required
+        ("--a", 2, SystemExit('expected 2 argument')),
+        ("--noa", 2, SystemExit('the following arguments are required')),
+        ("--a=true", 2, SystemExit('expected 2 argument')),
+        ("--a true false", 2, [True, False]),
+        # 1+ argument explicitly required
+        ("--a", '+', SystemExit('expected at least one argument')),
+        ("--noa", '+', SystemExit('the following arguments are required')),
+        ("--a=true", '+', [True]),
+        ("--a true false", '+', [True, False]),
+        # 0 or 1+ argument explicitly required
+        ("--a", '*', []),
+        ("--noa", '*', SystemExit('unrecognized arguments')),
+        ("--a=true", '*', [True]),
+        ("--a true false", '*', [True, False]),
+    ],
+)
+def test_bool_nargs(flag, nargs, a, capsys: pytest.CaptureFixture):
+    @dataclass
+    class MyClass(TestSetup):
+        """Some extension of base-class `Base`"""
+        a: bool = helpers.field(nargs=nargs)
```

Comment:
By setting `nargs` explicitly, you're bypassing the logic that generates the `nargs` from the annotation and context.
This might be fine, if you're just testing the new Action class on it's own. 
The only thing that bugs me is that the field annotation here is often incorrect (here with nargs="*" you should have a `list[bool]` annotation instead of `bool`).
It could be fine in some situations to manually specify `nargs`, for example if you have a `list[bool]` annotation, you could pass `nargs="+"` or `nargs > 1` to specify the number of values you expect.
Also, FYI, when using `tuple[bool, bool, bool]`, the generated `nargs=3`.

Reply:
> By setting nargs explicitly, you're bypassing the logic that generates the nargs from the annotation and context.

But this is a limitation of SimpleParsing right ? For example:

```python
@dataclass
class MyClass(TestSetup):
        a: int = helpers.field(nargs='*')
    
flags = MyClass.setup('--a 1 2 3')
assert flags.a == [1, 2, 3]
```

So I don't think this is related to this PR.

Reply:
It's not exactly a limitation of `simple-parsing`. it's just that the annotation you're using (`int`) becomes incorrect if you set `nargs="*"` explicitly.

If you had this, then it would be correct:
```python
@dataclass
class MyClass(TestSetup):
        a: list[int]

flags = MyClass.setup('--a 1 2 3')
assert flags.a == [1, 2, 3]
```

Reply:
I just meant if we want to fix this behavior (e.g. prevent `nargs='+'` for non-list types), this should be done in another PR as this behavior is not related to this specific action.

Reply:
Fair enough!